spec/graphql/mutations/add_location_spec.rb

```diff
@@ -0,0 +1,102 @@
+require 'rails_helper'
+
+RSpec.describe 'addLocation mutation', type: :request do
+  before :each do
+    @existing_user = create(:user)
+  end
+
+  describe 'with a valid Google token and valid location' do
+    it 'creates a new location' do
+      VCR.use_cassette('add_location_mutation_spec/valid_location') do
+        mutation = add_valid_location_mutation
+
+        post '/graphql', params: {
+                           google_token: @existing_user.google_token,
+                           query: mutation
+                         }
+
+        json = JSON.parse(response.body, symbolize_names: true)
+
+        data = json[:data][:addLocation]
+
+        expect(data[:message]).to eq('Location successfully added')
+      end
+    end
+  end
+
+  describe 'with a valid Google token and invalid location' do
+    it 'does not create a new location' do
+      VCR.use_cassette('add_location_mutation_spec/invalid_location') do
+        mutation = add_invalid_location_mutation
+
+        post '/graphql', params: {
+                           google_token: @existing_user.google_token,
+                           query: mutation
+                         }
+        json = JSON.parse(response.body, symbolize_names: true)
+
+        data = json[:data][:addLocation]
+        error_message = json[:errors][0][:message]
+
+        expect(data).to be_nil
+        expect(error_message).to eq('Invalid address entered')
+      end
+    end
+  end
+
+  describe 'with an invalid Google token' do
+    it 'does not create a new location' do
+      user = double(
```

Comment:
We don't really need to define a user in this case -- we can just put a bogus google_token in the params.